### PR TITLE
docs($http): Add explanation for error status

### PR DESCRIPTION
**Pull request by AbhishekGarg**
Add explanation for error status codes in $http call and line about normalization to zero in case of negative status code.
This alleviates confusion about status codes for which error callback would be called and final values to be expected from an erroneous call.

### CODE REVIEW

**Comment by gkalpak** [comment]
@AbhishekGarg, we need a signed CLA before we can merge this.
Also, please add `Fixes #11945` at the end of your commit message, so that issue it closed when/if this is merged.

**Comment by AbhishekGarg** [comment]
I have signed the CLA.
@gkalpak I'll make the suggested changes and changes for successful build asap.

**Comment by AbhishekGarg** [comment]
@gkalpak Updated commit message and changed for travis-ci check.
Could you please have a look?

File: src/ng/http.js [hunk 1]
```diff
@@ -451,9 +451,11 @@ function $HttpProvider() {
      *   - **statusText** – `{string}` – HTTP status text of the response.
      *
      * A response status code between 200 and 299 is considered a success status and
-     * will result in the success callback being called. Note that if the response is a redirect,
-     * XMLHttpRequest will transparently follow it, meaning that the error callback will not be
-     * called for such responses.
+     * will result in the success callback being called. Response status code other than
+     * the mentioned ones is considered as error status. The error callback is called in such case.
+     * Also, any negative status code is normalized to zero. Other codes are returned as such.
```
Comment (gkalpak) [inline src/ng/http.js]: Only codes < -1 are normalized. -1 is returned as is (and usually, means aborted, e.g. via timeout).

**Comment by gkalpak** [comment]
I tweaked it a bit and merged. Thx !

**Pull request closed without merging**